node/pkg/vaa/structs_test.go

```diff
@@ -0,0 +1,117 @@
+package vaa
+
+import (
+	"encoding/hex"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestChainIDFromString(t *testing.T) {
+	type test struct {
+		i string
+		o ChainID
+	}
+
+	// Positive Test Cases
+	p_tests := []test{
+		{i: "solana", o: ChainIDSolana},
+		{i: "ethereum", o: ChainIDEthereum},
+		{i: "terra", o: ChainIDTerra},
+		{i: "bsc", o: ChainIDBSC},
+		{i: "polygon", o: ChainIDPolygon},
+		{i: "avalanche", o: ChainIDAvalanche},
+		{i: "oasis", o: ChainIDOasis},
+		{i: "fantom", o: ChainIDFantom},
+		{i: "algorand", o: ChainIDAlgorand},
+		{i: "ethereum-ropsten", o: ChainIDEthereumRopsten},
+		{i: "Solana", o: ChainIDSolana},
+		{i: "Ethereum", o: ChainIDEthereum},
+		{i: "Terra", o: ChainIDTerra},
+		{i: "Bsc", o: ChainIDBSC},
+		{i: "Polygon", o: ChainIDPolygon},
+		{i: "Avalanche", o: ChainIDAvalanche},
+		{i: "Oasis", o: ChainIDOasis},
+		{i: "Fantom", o: ChainIDFantom},
+		{i: "Algorand", o: ChainIDAlgorand},
+		{i: "Karura", o: ChainIDKarura},
+		{i: "Acala", o: ChainIDAcala},
+	}
+
+	// Negative Test Cases
+	n_tests := []test{
+		{i: "Unknown", o: ChainIDUnset},
+	}
+
+	for _, tc := range p_tests {
+		chainId, err := ChainIDFromString(tc.i)
+		assert.Equal(t, tc.o, chainId)
+		assert.Nil(t, err)
+	}
+
+	for _, tc := range n_tests {
+		chainId, err := ChainIDFromString(tc.i)
+		assert.Equal(t, tc.o, chainId)
+		assert.NotNil(t, err)
+	}
+}
+
+func TestAddress_MarshalJSON(t *testing.T) {
+	addr := Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+	expected := "223030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303422"
+	marshalJsonAddress, err := addr.MarshalJSON()
+	assert.Equal(t, hex.EncodeToString(marshalJsonAddress), expected)
+	assert.Nil(t, err)
+}
+
+func TestAddress_String(t *testing.T) {
+	addr := Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+	expected := "0000000000000000000000000000000000000000000000000000000000000004"
+	assert.Equal(t, addr.String(), expected)
+}
+
+func TestAddress_Bytes(t *testing.T) {
+	addr := Address{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4}
+	expected := []byte{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4}
+	assert.Equal(t, addr.Bytes(), expected)
+}
+
+func TestSignatureData_MarshalJSON(t *testing.T) {
+	sigData := SignatureData{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0}
+	marshalJsonSigData, err := sigData.MarshalJSON()
```

Comment:
if this err does actually return, the assert.Equal() will blow up since you can't `hex.EncodeToString()` on a `nil` type.

For this, testify has [require](https://pkg.go.dev/github.com/stretchr/testify/require).

The test could be rewritten as:

```
	sigData := SignatureData{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0}
	marshalJsonSigData, err := sigData.MarshalJSON()
	require.Nil(t, err)

	expected := "223030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303430303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303030303034303022"
	assert.Equal(t, hex.EncodeToString(marshalJsonSigData), expected)
```

If the `require.Nil()` fails, then the rest of the test will not execute and you'll save the output of a blow up and error in the tests.

Reply:
Thanks!, was unaware of require.